### DISCUSSION TITLE
How to convert Array<T> to Vec<T>?

### DISCUSSION THREAD

**Discussion by 3togo** [arrayfire/arrayfire-rust#359] [Q&A] [answered]
How to convert `Array<T>` to `Vec<T>`?

**Comment by 3togo** [comment, marked as answer]
fn to_vec<T:HasAfEnum+Default+Clone>(array:&Array<T>) -> Vec<T> {
    let mut vec = vec!(T::default();array.elements());
    array.host(&mut vec);
    return vec;
}

After calling function host, will vec still in gpu memory?

**Reply by 9prady9** [reply]
host method only copies the contents from GPU to CPU. As long as the associated rust object( or af_array C handle) is live, the GPU memory stays. Host call doesn't remove the contents from GPU.

**Reply by 3togo** [reply]
Many thanks
